adapter/noop2/noop_test.go

```diff
@@ -79,82 +83,72 @@ func TestBasic(t *testing.T) {
 		t.Errorf("Got error %v, expecting success", err)
 	}
 
-	handler, err := builder.Build(nil, nil)
-	if err != nil {
-		t.Errorf("Got error %v, expecting success", err)
-	}
-
-	checkNothingHandler := handler.(checknothing.Handler)
-	result, caching, err := checkNothingHandler.HandleCheckNothing(nil)
-	if err != nil {
+	var handler adapter.Handler
+	if h, err := builder.Build(nil, nil); err != nil {
 		t.Errorf("Got error %v, expecting success", err)
+	} else {
+		handler = h
 	}
 
-	if !result {
-		t.Errorf("Got false, expecting true result")
+	var exptCheckRes = adapter.CheckResult{
+		Status:        rpc.Status{Code: int32(rpc.OK)},
+		ValidDuration: 1000000000 * time.Second,
+		ValidUseCount: 1000000000,
 	}
-
-	if caching.ValidUseCount < 1000 {
-		t.Errorf("Got use count of %d, expecting at least 1000", caching.ValidUseCount)
+	var exptReportResult = adapter.ReportResult{
+		Status: rpc.Status{Code: int32(rpc.OK)},
 	}
 
-	if caching.ValidDuration < 1000*time.Second {
-		t.Errorf("Got duration of %v, expecting at least 1000 seconds", caching.ValidDuration)
+	checkNothingHandler := handler.(checknothing.Handler)
+	if result, err := checkNothingHandler.HandleCheckNothing(context.TODO(), nil); err != nil {
+		t.Errorf("Got error %v, expecting success", err)
+	} else if !reflect.DeepEqual(result, exptCheckRes) {
```

Comment:
The API contract isn't that noop needs to return this exact value, that's why I prefer using explicit > comparisons instead of full equality checks.